paymentsheet/src/main/java/com/stripe/android/paymentsheet/LinkHandler.kt

```diff
@@ -113,6 +95,7 @@ internal class LinkHandler @Inject constructor(
 
         if (state == null) return
 
+        linkConfigurationCoordinator.setConfiguration(state.configuration)
```

Comment:
I don't intend the `setConfiguration` function. This was mainly added in this PR to solve the `FlowController` use case which after moving account status fetching caused `InlineSignup` to not be shown.

This happened because `LinkHandler` used to always initialize `LinkConfigurationCoordinator` when fetching the account status. Calling any function of `LinkConfigurationCoordinator` causes the coordinator to get or create a `LinkComponent` to use. `LinkInlineSignup` & `LinkOptionalInlineSignup` composables attempt to use the stored `component` which needs to be initialized first by making a function call to `getAccountStatusFlow`, `signInWithUserInput`, or `attachNewCardToAccount`.

It's pretty assumptious on the part of the composables to assume these are initialized beforehand. In the follow-up PR to move `LinkElement`, I'm going to change `setComponent` to be `getComponent` and pass the config through to `LinkFormElement` instead and fetch a component directly (either by creating one or getting the stored one if it exists).